File: frontend/lib/models/poll.dart
import 'option.dart';
import 'user.dart';

class Poll {
  final int id;
  final String question;
  final int createdBy;
  final int status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<Option> options;

  Poll({
    required this.id,
    required this.question,
    required this.createdBy,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    required this.options,
  });

  factory Poll.fromJson(Map<String, dynamic> json) {
    var optionsFromJson = json['options'] as List;
    List<Option> optionList =
        optionsFromJson.map((i) => Option.fromJson(i)).toList();

    return Poll(
      id: json['id'],
      question: json['question'],
      createdBy: json['created_by'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      options: optionList,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'question': question,
      'created_by': createdBy,
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'options': options.map((option) => option.toJson()).toList(),
    };
  }
}

File: backend/models/Poll.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
    const Poll = sequelize.define('Poll', {
        id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        question: {
            type: DataTypes.STRING,
            allowNull: false
        },
        created_by: {
            type: DataTypes.INTEGER,
            allowNull: false,
            references: {
                model: 'users',
                key: 'id',
            }
        },
        status: {
            type: DataTypes.STRING,
            allowNull: false
        },
        created_at: {
            type: DataTypes.DATE,
            allowNull: false
        },
        updated_at: {
            type: DataTypes.DATE,
            allowNull: true
        }
    }, {
        timestamps: true,
        createdAt: 'created_at',
        updatedAt: 'updated_at',
        tableName: 'poll'
    });

    Poll.associate = function(models) {
      Poll.belongsTo(models.User, {foreignKey: 'created_by', as: 'creator'});
      Poll.hasMany(models.Option, {foreignKey: 'poll_Id', as: 'options'});
    };

    return Poll;
};
